CodeEdit/Commands/CommandPaletteView.swift

```diff
@@ -0,0 +1,285 @@
+        }
+        .background(EffectView(.sidebar, blendingMode: .behindWindow))
+        .foregroundColor(.gray)
+        .edgesIgnoringSafeArea(.vertical)
+        .frame(minWidth: 600,
+           minHeight: self.state.isShowingCommandsList ? 400 : 28,
+           maxHeight: self.state.isShowingCommandsList ? .infinity : 28)
+    }
+}
+
+struct CommandPaletteView_Previews: PreviewProvider {
+    static var previews: some View {
+        CommandPaletteView(
+            state: .init(),
+            closePalette: {}
+        )
+    }
+}
+
+class ActionAwareInputView: NSTextView, NSTextFieldDelegate {
+
+    var onDown: ((NSEvent) -> Bool)?
+    var onTextChange: ((String) -> Void)?
+
+    func control(_ control: NSControl, textView: NSTextView, doCommandBy commandSelector: Selector) -> Bool {
+        return true
+    }
+
+    override var acceptsFirstResponder: Bool { return true }
+
+    override public func keyDown(with event: NSEvent) {
+        if onDown!(event) {
+            // We don't want to pass event down the pipe if it was handled.
+            // By handled I mean its keycode was used for something else than typing
+            return
+        }
+
+        super.keyDown(with: event)
+    }
+
+    override public func didChangeText() {
+        onTextChange?(self.string)
+    }
+
+}
+
+public struct SearchResultLabel: NSViewRepresentable {
+
+    var labelName: String
+    var textToMatch: String
+    var fontColor: Color
+
+    public func makeNSView(context: Context) -> some NSTextField {
+        let label = NSTextField(wrappingLabelWithString: labelName)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.drawsBackground = false
+        label.textColor = NSColor(fontColor.opacity(0.55))
+        label.isEditable = false
+        label.isSelectable = false
+        label.layer?.cornerRadius = 10.0
+        label.font = .labelFont(ofSize: 13)
+        label.allowsDefaultTighteningForTruncation = false
+        label.cell?.truncatesLastVisibleLine = true
+        label.cell?.wraps = true
+        label.maximumNumberOfLines = 1
+        label.attributedStringValue = highlight()
+        return label
+    }
+
+    func highlight() -> NSAttributedString {
+
+        let attribText = NSMutableAttributedString(string: self.labelName)
+        let range: NSRange = attribText.mutableString.range(of: self.textToMatch,
+                                                                options: NSString.CompareOptions.caseInsensitive)
+
+        var attributes: [NSAttributedString.Key: Any] = [:]
+        attributes[.foregroundColor] = NSColor(fontColor.opacity(0.85))
+        attribText.addAttributes(attributes, range: range)
+        return attribText
+    }
+
+    public func updateNSView(_ nsView: NSViewType, context: Context) {
+        nsView.textColor = NSColor(fontColor.opacity(0.55))
+        nsView.attributedStringValue = highlight()
+    }
+
+}
+
+public struct ActionAwareInput: NSViewRepresentable {
```

Comment:
Please add proper documentation to `public` classes, structs, methods and variables/constants.

A guide about code documentation can be found [here](https://nshipster.com/swift-documentation/).